tokensRemover/metaDataRemover/txCreator.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/blockchain"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/builders"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/interactors"
+	"github.com/ElrondNetwork/elrond-tools-go/tokensRemover/metaDataRemover/config"
+)
+
+func createShardTxs(
+	outFile string,
+	cfg *config.Config,
+	shardPemsDataMap map[uint32]*pkAddress,
+	shardTxsDataMap map[uint32][][]byte,
+) error {
+	if len(shardPemsDataMap) != len(shardTxsDataMap) {
+		return fmt.Errorf("provided invalid input; expected number of pem files = number of shards in tokens input; got num shard tokens = %d, num pem files = %d",
+			len(shardPemsDataMap), len(shardTxsDataMap))
+	}
+
+	args := blockchain.ArgsElrondProxy{
+		ProxyURL:            cfg.ProxyUrl,
+		CacheExpirationTime: time.Minute,
+		EntityType:          core.Proxy,
+	}
+
+	proxy, err := blockchain.NewElrondProxy(args)
+	if err != nil {
+		return err
+	}
+
+	txBuilder, err := builders.NewTxBuilder(blockchain.NewTxSigner())
+	if err != nil {
+		return err
+	}
+
+	ti, err := interactors.NewTransactionInteractor(proxy, txBuilder)
+	if err != nil {
+		return err
+	}
+
+	err = createOutputFileIfDoesNotExist(outFile)
+	if err != nil {
+		return err
+	}
+
+	for shardID, txsData := range shardTxsDataMap {
+		pemData, found := shardPemsDataMap[shardID]
+		if !found {
+			return fmt.Errorf("no pem data provided for shard = %d", shardID)
+		}
+
+		log.Info("starting to create txs", "shardID", shardID, "num of txs", len(txsData))
+		txsInShard, err := createTxs(pemData, proxy, ti, txsData, cfg.GasLimit)
+		if err != nil {
+			return err
+		}
+
+		file := outFile + "/txsShard" + strconv.Itoa(int(shardID)) + ".json"
```

Comment:
one useful feature would be to have a configured number of txs per bulk. or even better, a num-nonces-based txs split. For example, we could have txsShard2_bulk0.json, txsShard2_bulk1.json, and so on. each bulk could contain a number of pre-defined txs, or even better, each file will contain a number of nonces (200 for example). This will make the sending of the transactions much easier, since if we would have a file containing all txs, it would be really hard to determine how to send them correctly

Reply:
I disagree with this.
The tool that will handle txs sending could format its input however it suits for it. This tool should not hold extra logic to format its output specific to the input of another tool. We simply output all signed txs for each shard, which is is the most complete information to be formatted as pleased from this point forward

> one useful feature would be to have a configured number of txs per bulk.

just to add extra clarification, in the context:  one bulk == one tx, which holds multiple tokens with multiple nonces

> or even better, a num-nonces-based txs split.

No, we do **not** want to choose the number of nonces based on number of txs (this is exactly the reverted logic of how it is implemented now). It does not matter how many txs are created, we only care for consistency in number of nonces to be deleted per tx.

> For example, we could have txsShard2_bulk0.json, txsShard2_bulk1.json, and so on. each bulk could contain a number of pre-defined txs, or even better, each file will contain a number of nonces (200 for example). This will make the sending of the transactions much easier, since if we would have a file containing all txs, it would be really hard to determine how to send them correctly

I'm not sure I understand the benefit of this approach. This adds a lot of overhead. Why not have a tool that, for example:
1. Reads all txs from shardX in memory
2. Simply sends one tx per block (or even multiple, although I doubt it) and keeps an internal counter for each sent tx
3. Store this counter in a file
4. In case anything happens during execution (e.g.: internet connection lost, tool forced stop), we could restart it and start from the last stored index

Reply:
okay, can remain as it is :+1: